### PR TITLE
ARROW-874: [JS] Read dictionary-encoded vectors

### PR DESCRIPTION
**Pull request by TheNeuralBit**

### CODE REVIEW

**Comment by TheNeuralBit** [comment]
Includes #633 and #642

**Comment by wesm** [comment]
Will try to get a review up on this today

**Comment by TheNeuralBit** [comment]
No rush - sorry to flood you with PRs just as 0.3 is going out

File: js/lib/arrow.ts [hunk 1]
```diff
@@ -81,27 +219,174 @@ function _loadFooter(bb) {
...
+        org.apache.arrow.flatbuf.FloatingPoint.addPrecision(builder, type.precision());
+        typeOffset = org.apache.arrow.flatbuf.FloatingPoint.endFloatingPoint(builder);
+    } else if (typeType === Type.Utf8) {
+        org.apache.arrow.flatbuf.Utf8.startUtf8(builder);
+        typeOffset = org.apache.arrow.flatbuf.Utf8.endUtf8(builder);
+    } else if (typeType === Type.Date) {
+        var type = field.type(new org.apache.arrow.flatbuf.Date());
+        org.apache.arrow.flatbuf.Date.startDate(builder);
+        org.apache.arrow.flatbuf.Date.addUnit(builder, type.unit());
+        typeOffset = org.apache.arrow.flatbuf.Date.endDate(builder);
```
Comment (wesm) [inline js/lib/arrow.ts]: We'll want to encapsulate this in a reusable function

**Review by wesm — APPROVED** [review]
+1, nice to see dictionary reads working in JS! I'm merging this and we can get to work on testing and refactoring / API improvements in follow up patches

**Pull request closed without merging**